### ISSUE TITLE
visiting a url with a secondary domain like string causes cy.visit to incorrectly error

### ISSUE THREAD

**Issue by jheijkoop** [cypress-io/cypress#255]
With cypress `0.17.5` with a baseUrl set in cypress.json to `my-domain` my `.route('/#/foo/bar').as('alias')`'s stop matching and `.visit('/#/foo/bar')` gives the following error:
```
Error:     CypressError: cy.visit() failed because you are attempting to visit a second unique domain.
You may only visit a single unique domain per test.
Different subdomains are okay, but unique domains are not.
The previous domain you visited was: 'http://my-domain:5100'
You're attempting to visit this new domain: 'http://my-domain:5100/#/foo/bar?email=foo@bar.baz'
You may need to restructure some of your code to prevent this from happening.
https://on.cypress.io/cannot-visit-second-unique-domain
```
when calling `$location.path('/foo/bar').search({email: 'foo@bar.baz'});` from JS through triggering a click in a previous command and having Angular (1.4.9) firing an action.
```
cy.
  .get('#my-button`).click() // action redirects to '/#/foo/bar?email=foo@bar.baz' with angulars $location  
  .visit('/#/foo/bar?email=foo@bar.baz')
```

**Comment by brian-mann** [comment]
Can you paste the rest of your test code? I want to see what your first visit is, and what your baseUrl is set in `cypress.json`.

This looks like a bug though - but I cannot recreate it locally, so that means something must be different about what you're doing. We have a ton of test coverage specifically around this functionality so it is likely and edge case.

A screenshot of the failure + your command log would also be helpful, but test code would be ideal. If possible can you reduce this down to the simplest set of commands which can recreate the problem?

**Comment by jheijkoop** [comment]
it seems it might be something different. The url I'm visiting now is of the form `/foo/bar?id=foo@bar.baz` and if I remove the `?foo@bar.baz` part I no longer get the domain conflict error message with `0.17.6`

**Comment by brian-mann** [comment]
Confirmed, the problem is that calling `cy.visit` with a string that has another "domain-like string" in it causes `cy.visit` to assume the wrong origin which triggers the 2nd origin error message.

<img width="508" alt="screen shot 2016-10-09 at 6 53 43 pm" src="https://cloud.githubusercontent.com/assets/1268976/19222140/c8cf550e-8e51-11e6-900a-2125dc820cd6.png">

**Comment by brian-mann** [comment]
Fixed in [`0.17.7`](https://github.com/cypress-io/cypress/wiki/Changelog#0177-10122016).
